ttnn/cpp/ttnn/operations/data_movement/repeat/repeat.cpp

```diff
@@ -18,10 +21,18 @@ ttnn::Tensor RepeatOperation::invoke(
     const Shape & repeat_dims,
     const std::optional<MemoryConfig>& memory_config_arg) {
 
-    std::vector<Tensor> output_tensors = {Tensor(operation::get_workers_for_op_output({input_tensor}))};
+    // special path for handling padded tiled tensors that are naively repeated
+    // *with padding* in the current implementation
+    std::optional<Tensor> formatted_input_tensor = std::nullopt;
+    if (input_tensor.get_layout() != Layout::ROW_MAJOR
+        && input_tensor.get_legacy_shape().without_padding() != input_tensor.get_legacy_shape()) {
+        formatted_input_tensor = ttnn::untilize(input_tensor);
```

Comment:
Untilize works for the (1,1,1[32],1[32]) tensor? Does it spit out a (1,1,1,1,)?

Reply:
As far as I understand, untilize merely spits the tiles into a row major tensor. It's not aware of padding so you have to slice afterwards to get the unpadded component of the original tiles. I think I can drop this part though, TBD.